platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbDataSource.java

```diff
@@ -239,15 +241,20 @@ public MerkleDbDataSource(
         // create path to disk location index
         final boolean forceIndexRebuilding = database.getConfig().indexRebuildingEnforced();
         if (tableConfig.isPreferDiskBasedIndices()) {
-            pathToDiskLocationInternalNodes = new LongListDisk(dbPaths.pathToDiskLocationInternalNodesFile);
+            final FileSystemManager fileSystemManager =
+                    FileSystemManager.create(ConfigurationHolder.getInstance().get());
+            pathToDiskLocationInternalNodes =
+                    new LongListDisk(dbPaths.pathToDiskLocationInternalNodesFile, fileSystemManager);
         } else if (Files.exists(dbPaths.pathToDiskLocationInternalNodesFile) && !forceIndexRebuilding) {
             pathToDiskLocationInternalNodes = new LongListOffHeap(dbPaths.pathToDiskLocationInternalNodesFile);
         } else {
             pathToDiskLocationInternalNodes = new LongListOffHeap();
         }
         // path to disk location index, leaf nodes
         if (tableConfig.isPreferDiskBasedIndices()) {
-            pathToDiskLocationLeafNodes = new LongListDisk(dbPaths.pathToDiskLocationLeafNodesFile);
+            final FileSystemManager fileSystemManager =
```

Comment:
What's the reason for having multiple instance of `FileSystemManager`? Given the same instance of `Configuration` they will be identical.

Reply:
I missed that during the lifecycle of a `MerkleDbDatasource` instance the configuration instance will be the same hence the `FileSystemManager` will be identical as well.
Given that I agree with the suggestion from below to have one `FileSystemManager` on `MerkleDbDatasource` level and then pass it as an argument to the underlying data structures.

Reply:
Fixed.